javalib/src/main/scala/java/util/LinkedList.scala

```diff
@@ -0,0 +1,453 @@
+    }
+  }
+
+  override def contains(o: Any): Boolean = {
+    val boxed = Box(o.asInstanceOf[E])
+    var found = false
+    val iter = iterator()
+    while (iter.hasNext() && !found) {
+      val next = iter.next()
+      if (Box(next) === boxed) {
+        found = true
+      }
+    }
+
+    found
+  }
+
+  override def size(): Int =
+    _size
+
+  override def add(e: E): Boolean = {
+    addLast(e)
+    true
+  }
+
+  override def remove(o: Any): Boolean =
+    _removeOccurrence(listIterator, o)
+
+  override def addAll(c: Collection[_ <: E]): Boolean = {
+    val iter = c.iterator
+    val changed = iter.hasNext()
+    while (iter.hasNext())
+      addLast(iter.next())
+
+    changed
+  }
+
+  override def clear(): Unit = {
+    head = Node()
+    last = Node()
+    _size = 0
+  }
+
+  private def getNodeAt(index: Int): Node[Box[E]] = {
+    if (index == 0) head
+    else if (index == (size - 1)) last
+    else {
+      var current: Node[Box[E]] = null
+      if (index <= (size/2)) {
+        current = head
+        for (_ <- 0 until index) current = current.next
+      } else {
+        current = last
+        for (_ <- index until (size - 1)) current = current.prev
+      }
+      current
+    }
+  }
+
+  override def get(index: Int): E = {
+    checkIndexInBounds(index)
+    getNodeAt(index).value.inner
+  }
+
+  override def set(index: Int, element: E): E = {
+    checkIndexInBounds(index)
+    val node = getNodeAt(index)
+    val oldValue = node.value.inner
+    node.value = Box(element)
+    oldValue
+  }
+
+  private def addNode(next: Node[Box[E]], e: E): Unit = {
+    if (next.uuid == head.uuid) addFirst(e)
+    else if (next.uuid == null) addLast(e)
+    else {
+      var current = next
+
+      val node = Node(Box(e), prev = current.prev, next = current)
+      current.prev.next = node
+      current.prev = node
+
+      _size += 1
+    }
+  }
+
+  override def add(index: Int, element: E): Unit = {
+    checkIndexOnBounds(index)
+    if (index == 0) addFirst(element)
+    else if (index == size) addLast(element)
+    else addNode(getNodeAt(index), element)
+  }
+
+  private def removeNode(node: Node[Box[E]]): E = {
+    if (node.uuid == head.uuid) removeFirst()
+    else if (node.uuid == last.uuid) removeLast()
+    else {
+      var current = node
+
+      current.prev.next = current.next
+      current.next.prev = current.prev
+
+      _size -= 1
+
+      current.value.inner
+    }
+  }
+
+  override   def remove(index: Int): E = {
+    checkIndexInBounds(index)
+    if (index == 0) removeFirst()
+    else if (index == (size - 1)) removeLast()
+    else removeNode(getNodeAt(index))
+  }
+
+  override def indexOf(o: Any): Int = {
+    val boxed = Box(o.asInstanceOf[E])
+    var current: Node[Box[E]] = getNodeAt(0)
+    var count = 0
+    while ((current ne null) &&
+        !current.value.equals(boxed)) {
+      current = current.next
+      count += 1
+    }
+
+    Option(current).fold(-1)(_ => count)
```

Comment:
No need to override `indexOf`. The implementation in `AbstractLinkedList` is already good.